pkg/solana/state_cache.go

```diff
@@ -0,0 +1,165 @@
+package solana
+
+import (
+	"context"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	bin "github.com/gagliardetto/binary"
+	"github.com/gagliardetto/solana-go"
+	"github.com/gagliardetto/solana-go/rpc"
+	"github.com/smartcontractkit/chainlink/core/utils"
+
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana/client"
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana/config"
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana/logger"
+)
+
+var (
+	configVersion uint8 = 1
+)
+
+type StateCache struct {
+	utils.StartStopOnce
+	// on-chain program + 2x state accounts (state + transmissions)
+	ProgramID      solana.PublicKey
+	StateID        solana.PublicKey
+	StoreProgramID solana.PublicKey
+
+	stateLock sync.RWMutex
```

Comment:
idiomatic to use non-pointers for the locks then just use pointers to the object holding the lock. Then you can make use of its default unlocked value